code/__HELPERS/AStar.dm

```diff
@@ -61,117 +73,114 @@ Actual Adjacent procs :
 	return b.f - a.f
 
 //wrapper that returns an empty list if A* failed to find a path
-/proc/get_path_to(caller, end, dist, maxnodes, maxnodedepth = 30, mintargetdist, adjacent = /turf/proc/reachableAdjacentTurfs, id=null, turf/exclude=null, simulated_only = 1)
+/proc/get_path_to(caller, end, dist, maxnodes, maxnodedepth = 30, mintargetdist, adjacent = /turf/proc/reachableTurftest, id=null, turf/exclude=null, simulated_only = 1)
+	var/l = SSpathfinder.getfree()
+	while(!l)
+		stoplag(3)
+		l = SSpathfinder.getfree()
 	var/list/path = AStar(caller, end, dist, maxnodes, maxnodedepth, mintargetdist, adjacent,id, exclude, simulated_only)
+
+	SSpathfinder.found(l)
 	if(!path)
 		path = list()
+
 	return path
 
-//the actual algorithm
-/proc/AStar(caller, end, dist, maxnodes, maxnodedepth = 30, mintargetdist, adjacent = /turf/proc/reachableAdjacentTurfs, id=null, turf/exclude=null, simulated_only = 1)
-	var/list/pnodelist = list()
+/proc/AStar(caller, var/turf/end, dist, maxnodes, maxnodedepth = 30, mintargetdist, adjacent = /turf/proc/reachableTurftest, id=null, turf/exclude=null, simulated_only = 1)
 	//sanitation
-	var/start = get_turf(caller)
-	if(!start)
+	var/turf/start = get_turf(caller)
+	if((!start)||(start.z != end.z)||(start == end)) //no PF between z levels
```

Comment:
pathfinding?

Reply:
yes

Reply:
can you expand it to say PF, the extra words don't hurt anyone and makes the comment clearer